### DISCUSSION TITLE
v3.0.0-preview.7

### DISCUSSION THREAD

**Discussion by SteveL-MSFT** [PowerShell/DSC#416] [Announcements]
## What's Changed
* (DOCS) Options refactor by @michaeltlombardi in https://github.com/PowerShell/DSC/pull/326
* update to next version by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/331
* add progress bar by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/323
* rename `provider` to `adapter` by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/334
* add `kind` to dscresource by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/338
* use tracing indicatif to sync tracing and progress bar rendering by @tgauth in https://github.com/PowerShell/DSC/pull/343
* rename group resources to conform to namespace standard by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/341
* enable building msix package by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/340
* Add run command resource by @tgauth in https://github.com/PowerShell/DSC/pull/321
* pre-allocate string for table writer by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/345
* Updates to DSC_CONFIG_ROOT env var by @anmenaga in https://github.com/PowerShell/DSC/pull/342
* Add metadata so adapter resource knows if input is configuration or for a resource by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/348
* (DOCS) Update docs for `alpha.5` by @michaeltlombardi in https://github.com/PowerShell/DSC/pull/350
* use bit flags for methods in table view by @tgauth in https://github.com/PowerShell/DSC/pull/346
* add div function by @tgauth in https://github.com/PowerShell/DSC/pull/349
* Add mul function by @tgauth in https://github.com/PowerShell/DSC/pull/347
* Enable metadata in config to declare if admin is required or not by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/351
* RebootPending resource by @anmenaga in https://github.com/PowerShell/DSC/pull/344
* implement sub function by @tgauth in https://github.com/PowerShell/DSC/pull/353
* implement add function by @tgauth in https://github.com/PowerShell/DSC/pull/352
* Fixed missing PS resources by @anmenaga in https://github.com/PowerShell/DSC/pull/355
* Add capabilities to DscResource schema by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/357
* Fix `envvar()` function to error if env var doesn't exist by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/358
* Add mod function by @tgauth in https://github.com/PowerShell/DSC/pull/360
* Fix indicatif prog bar by @tgauth in https://github.com/PowerShell/DSC/pull/359
* Add `reference()` function to get output from a resource by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/354
* Support for resource tracing info by @anmenaga in https://github.com/PowerShell/DSC/pull/287
* add min function by @tgauth in https://github.com/PowerShell/DSC/pull/361
* move file_lib to archive by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/367
* Add validation of resource output by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/362
* Enable passing of secure parameters by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/364
* Update release pipeline by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/370
* update version to preview.7 by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/372
* Rename `requires` in DscResource schema to `requireAdapter` by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/373
* Fix table view for column `Requires` to `RequireAdapter` by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/374
* add max function by @tgauth in https://github.com/PowerShell/DSC/pull/375
* Added adapter filter by @anmenaga in https://github.com/PowerShell/DSC/pull/377
* add int function by @tgauth in https://github.com/PowerShell/DSC/pull/376
* Rewrite registry resource to support delete operation by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/379
* add better error if path is accidentally passed to `-d` by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/380
* Update engine to call `delete` for resources that don't support `_exist` directly by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/382
* improve `no schema` error message by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/384
* update tree-sitter to 0.22 by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/386
* Validate semver of resource manifests by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/387
* Only emit source information for trace level `debug` and `trace` by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/388
* (MAINT) Skip builds on docs-only PRs by @michaeltlombardi in https://github.com/PowerShell/DSC/pull/392
* PowerShell adapter PR by @mgreenegit in https://github.com/PowerShell/DSC/pull/363
* Refactor arguments to command resources an mixed strings and objects for extensibility by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/385
* update .gitignore and add .gitattributes for tree-sitter by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/404
* add metadata to configuration result output by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/405
* Update osinfo_parameters.dsc.yaml by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/408
* (MAINT) Update GitHub templates by @michaeltlombardi in https://github.com/PowerShell/DSC/pull/393
* (DOCS) Document new and changed config functions by @michaeltlombardi in https://github.com/PowerShell/DSC/pull/391
* (SCHEMAS) Update schemas for preview.7 by @michaeltlombardi in https://github.com/PowerShell/DSC/pull/397
* Set/Test/Export and other updates for PSAdapter by @anmenaga in https://github.com/PowerShell/DSC/pull/411
* Release pipeline updates by @SteveL-MSFT in https://github.com/PowerShell/DSC/pull/413

**Full Changelog**: https://github.com/PowerShell/DSC/compare/v3.0.0-alpha.5...v3.0.0-preview.7

## Hashes of Packages

- DSC-3.0.0-preview.7-aarch64-apple-darwin.tar.gz
  - 30002C85DC2066065A07D0C16C4161C0D41AB66BF201AB4561951DFDC03A30D8
- DSC-3.0.0-preview.7-aarch64-pc-windows-msvc.zip
  - 47AF2F667FA975C6A1C70672DE142D94831716ECB57DE26E39BC0BC6E21AC7B7
- DSC-3.0.0-preview.7-aarch64-unknown-linux-gnu.tar.gz
  - 7BDE4549541B33A0CE89D867102D60F58B6C9C7A0616911052CE52DB0A58A1C5
- DSC-3.0.0-preview.7-x86_64-apple-darwin.tar.gz
  - 28C39FCBB78C088AA1C2E67E774108EE9E422D168C92D5B8A50C8EEEAF98DBD2
- DSC-3.0.0-preview.7-x86_64-pc-windows-msvc.zip
  - AA50F6BA87099A0F2337E7BD89C2C594563ED3594F3877E02596A4874FB2C722
- DSC-3.0.0-preview.7-x86_64-unknown-linux-gnu.tar.gz
  - 909067B93D21D676B3056D885C807FAF0482C453430612B6B28E16A42B2E14F4

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/DSC/releases/tag/v3.0.0-preview.7'>v3.0.0-preview.7</a>.</em>

**Comment by egobrane** [comment]
Good to see the team is hard at work on this. I haven't delved into DSC3.0 much but use 1.1 extensively. We are rolling out macbooks to our staff and I'm curious to see what level of automation and control DSC3.0 can eventually provide.

**Reply by 100110010111** [reply]
Does DSCv3 work for on-prem or is v1.1 as good as it gets?

**Reply by egobrane** [reply]
It looks like it does and can be monitored with Azure machine configuration for Arc-enabled "servers", but I haven't done enough experimentation to say for certain.
